src/main/java/com/google/api/gax/grpc/StreamingApiCallable.java

```diff
@@ -87,19 +85,57 @@ public void bind(ManagedChannel channel) {
    */
   public StreamObserver<RequestT> bidiStreamingCall(StreamObserver<ResponseT> responseObserver) {
     Preconditions.checkNotNull(channel);
-    return callable.bidiStreamingCall(responseObserver, CallContext.DEFAULT.withChannel(channel));
+    return callable.bidiStreamingCall(
+        responseObserver, CallContext.createDefault().withChannel(channel));
   }
 
   /**
-   * Conduct a server streaming call
+   * Conduct a bidirectional streaming call with the given {@link
+   * com.google.api.gax.grpc.CallContext}.
    *
    * @param responseObserver {@link io.grpc.stub.StreamObserver} to observe the streaming responses
+   * @param context {@link com.google.api.gax.grpc.CallContext} to provide context information of
+   *     the gRPC call. The existing channel will be overridden by the channel contained in the
+   *     context (if any).
+   * @return {@link StreamObserver} which is used for making streaming requests.
+   */
+  public StreamObserver<RequestT> bidiStreamingCall(
+      StreamObserver<ResponseT> responseObserver, CallContext context) {
+    if (context.getChannel() == null) {
+      context = context.withChannel(channel);
+    }
```

Comment:
Well actually, the unary case could to it exactly as you do, and bind the channel in UnaryApiCallable, because it is the top-level entry point, and there is no reason for the channel to change once the call has been issued. I would rather you make them consistent in this regard than do one differently.